File: examples/widgets/tileview/lv_example_tileview_1.py
#
# Create a 2x2 tile view and allow scrolling only in an "L" shape.
# Demonstrate scroll chaining with a long list that
# scrolls the tile view when it can't be scrolled further.
#
tv = lv.tileview(lv.scr_act())

# Tile1: just a label
tile1 = tv.add_tile(0, 0, lv.DIR.BOTTOM)
label = lv.label(tile1)
label.set_text("Scroll down")
label.center()

# Tile2: a button
tile2 = tv.add_tile(0, 1, lv.DIR.TOP | lv.DIR.RIGHT)

button = lv.button(tile2)

label = lv.label(button)
label.set_text("Scroll up or right")

button.set_size(lv.SIZE_CONTENT, lv.SIZE_CONTENT)
button.center()

# Tile3: a list
tile3 = tv.add_tile(1, 1, lv.DIR.LEFT)
list = lv.list(tile3)
list.set_size(lv.pct(100), lv.pct(100))

list.add_button(None, "One")
list.add_button(None, "Two")
list.add_button(None, "Three")
list.add_button(None, "Four")
list.add_button(None, "Five")
list.add_button(None, "Six")
list.add_button(None, "Seven")
list.add_button(None, "Eight")
list.add_button(None, "Nine")
list.add_button(None, "Ten")

File: examples/widgets/tileview/lv_example_tileview_1.c
#include "../../lv_examples.h"
#if LV_USE_TILEVIEW && LV_BUILD_EXAMPLES

/**
 * @title L-shaped tile view with scroll chaining
 * @brief Three tiles in an L layout where a ten-item list chains its scroll to the tile view.
 *
 * `lv_tileview_add_tile` places a label tile at (0, 0) allowing
 * `LV_DIR_BOTTOM`, a button tile at (0, 1) allowing `LV_DIR_TOP` and
 * `LV_DIR_RIGHT`, and a list tile at (1, 1) allowing `LV_DIR_LEFT`.
 * The button tile holds a centered `lv_button` labeled
 * `Scroll up or right`; the list tile holds a full-size `lv_list`
 * with ten buttons `One` through `Ten` whose scroll chains back into
 * the tile view when the list reaches its edge.
 */
void lv_example_tileview_1(void)
{
    lv_obj_t * tv = lv_tileview_create(lv_screen_active());

    /*Tile1: just a label*/
    lv_obj_t * tile1 = lv_tileview_add_tile(tv, 0, 0, LV_DIR_BOTTOM);
    lv_obj_t * label = lv_label_create(tile1);
    lv_label_set_text(label, "Scroll down");
    lv_obj_center(label);

    /*Tile2: a button*/
    lv_obj_t * tile2 = lv_tileview_add_tile(tv, 0, 1, (lv_dir_t)(LV_DIR_TOP | LV_DIR_RIGHT));

    lv_obj_t * btn = lv_button_create(tile2);

    label = lv_label_create(btn);
    lv_label_set_text(label, "Scroll up or right");

    lv_obj_set_size(btn, LV_SIZE_CONTENT, LV_SIZE_CONTENT);
    lv_obj_center(btn);

    /*Tile3: a list*/
    lv_obj_t * tile3 = lv_tileview_add_tile(tv, 1, 1, LV_DIR_LEFT);
    lv_obj_t * list = lv_list_create(tile3);
    lv_obj_set_size(list, LV_PCT(100), LV_PCT(100));

    lv_list_add_button(list, NULL, "One");
    lv_list_add_button(list, NULL, "Two");
    lv_list_add_button(list, NULL, "Three");
    lv_list_add_button(list, NULL, "Four");
    lv_list_add_button(list, NULL, "Five");
    lv_list_add_button(list, NULL, "Six");
    lv_list_add_button(list, NULL, "Seven");
    lv_list_add_button(list, NULL, "Eight");
    lv_list_add_button(list, NULL, "Nine");
    lv_list_add_button(list, NULL, "Ten");

}

#endif
